connectome_interpreter/utils.py

```diff
@@ -1087,3 +1087,164 @@ def path_for_ngl(path):
         df = df.drop_duplicates()
         dfs.append(df)
     return pd.concat(dfs)
+
+
+def make_grid_inputs(v1, v2, num_layers: int, grid_size: int = 10, timepoints: int | list = 0, device=None) -> Tuple[torch.Tensor, List[Tuple[float, float]]]:
```

Comment:
"Modern" numpy recommends using `numpy.typing.NDArray` rather than `np.ndarray` ([doc](https://numpy.org/doc/stable/reference/typing.html)):
```python
import numpy.typing as npt

def fun(xs: npt.NDArray, ...): ...
```